[#225 #232 #244] "Create new term" modal issues

bindeali opened the pull request:
## Info:
* Closes #225.
* Closes #232.
* Closes #244.
* Can be merged and closed right away.
## Detailed changes:
* Changed the look of new term error modal message.
* Changed the way the IRI is created. Anything that is not `/[a-zA-Z0-9À-ž-._~:/?#\[\]@!$&'()*+,;=]/` is removed from the IRI, but the check passes. The only thing character does not pass is `"`.
![image](https://user-images.githubusercontent.com/56399637/111866465-ea489d00-896d-11eb-8040-ce858642afb0.png)

* For testing purposes, I added a live view of what the IRI is going to be before saving.
![image](https://user-images.githubusercontent.com/56399637/111866416-93db5e80-896d-11eb-9650-c47f12b0e388.png)
* Implemented a label length check.
![image](https://user-images.githubusercontent.com/56399637/111866393-71494580-896d-11eb-92fe-01cb0726d2a1.png)

## Deploy previews: 
* [ML test](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-1101091271)
* [Datové fondy VS](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-1153182747)
* [Marie test](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-530560086)
* [mm-test](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-2000996011)
* [Sčítání lidu](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-2011762727)
* [Sbírka](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance1483547911)
* [Právnické a fyzické osoby v rejstřících](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-1872291546)
* [Správný model konkrétního číselníku](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance570218201)
* [čas](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-1474929552)
* [Karel Test](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-1231949029)
* [Školení - doména řidičů a vozidel](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-946737236)
* [Školení - doména poslanecké sněmovny](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance1938652144)
* [Školení - doména registru práv a povinností](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-510905468)
* [Školení - ukázka veřejné zakázky](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-936554450)
* [Školení - veřejné zakázky](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance1008211408)
* [Testovací WS pro formální jazyky](https://6065b556a93b7b0a44b8a9a4--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-2039280178)

psiotwo commented:
Checked #232 - works OK.
Checked #244 - works OK, wasn't able to reproduce the error.

psiotwo commented on src/function/FunctionCreateVars.ts:
```diff
@@ -16,7 +16,10 @@ export function createValues(values: { [key: string]: string[] }, prefixes: { [k
-        .trim().replace(/\s/g, '-').toLowerCase();
+        .trim()
+        .replace(/\s/g, '-')
+        .replace(/(?![a-zA-Z0-9À-ž-._~:/?#[\]@!$&'()*+,;=])./g, "")
```
The logic which is doing the same thing in TermIt is here:
https://github.com/opendata-mvcr/termit/blob/ac1424f8ec0d22df5d6cd1e1e36ce6857570d085/src/main/java/cz/cvut/kbss/termit/service/IdentifierResolver.java#L103

In general:
- architecture-wise - let's discuss, how to unify these and reuse the same logic - a common function library (**Assembly line toolkit**) with share functions, or a microservice. 
- content-wise - I don't think we should put too many restrictions on the IRIs - the more things we replace the more conflicts we get (e.g. "building & immovable property" vs. "building and immovable property" vs. "building immovable property")

bindeali commented:
Changed some things, explanation below:
* @karelklima Double quotes are allowed in the name and escaped.
* Changed the IRI ignored set to `(?&)"^<>`.
* Whitespaces and `\` in the label are changed to `-` for the IRI.
* Added a check to prevent IRI with an empty term name (such as `https://slovník.gov.cz/datový/sbírka/pojem/`)

I analyzed the way Termit normalizes IRIs. I noticed that it also removes `"^<>` characters, but such functionality isn't in the [code](https://github.com/opendata-mvcr/termit/blob/ac1424f8ec0d22df5d6cd1e1e36ce6857570d085/src/main/java/cz/cvut/kbss/termit/service/IdentifierResolver.java#L103) that @psiotwo provided. I eventually realized that Termit does not really remove these characters, but instead the following happens:

1. onChange of the label field on the New Term page sends a request to Termit server [/identifiers](https://app.swaggerhub.com/apis/ledvima1/TermIt/2.2.0#/default/post_rest_identifiers) to receive a normalized IRI.
2. If one of the aformentioned characters is present in the name, Termit server throws a 500 error and therefore the UI does not receive a new IRI - the IRI that the new term is saved under is such from before the error happens. For example, term name `foo"bar` get an IRI of `https://.../pojem/foo`.

Because of this, I still kept the IRI ignore set of characters, but made it as small as it can be. IMO this is a workable solution until we have this function in the toolkit implemented, but this is of course up to you.
